src/workspace.ts

```diff
@@ -77,32 +77,57 @@ export class WorkspaceItem extends TreeItem {
 	}
 }
 
+export function clearWorkspace(): void {
+	void window.showInformationMessage(
+		"Are you sure you want to clear the workspace? This cannot be reversed.",
+		"Confirm",
+		"Cancel"
+	).then(selection => {
+		if (selection == "Confirm") {
+			void runTextInTerm(`rm(list = ls())`)
+		}
+	})
+}
+
 export function saveWorkspace(): void {
-	window.showSaveDialog({
-		defaultUri: Uri.file('workspace.RData'),
+	void window.showSaveDialog({
+		defaultUri: Uri.file(getWorkspacePath() + path.sep + 'workspace'),
 		filters: {
 			'Data': ['RData']
-		}
+		},
+		title: 'Save workspace'
 	}
 	).then((uri: Uri | undefined) => {
 		if (uri) {
-			runTextInTerm(
-				`save.image(\"${(uri.fsPath.split(path.sep).join(path.posix.sep))}\")`
+			void runTextInTerm(
+				`save.image("${(uri.fsPath.split(path.sep).join(path.posix.sep))}")`
 			);
 		}
 	});
 }
 
 export function loadWorkspace(): void {
-	window.showOpenDialog({
+	void window.showOpenDialog({
+		defaultUri: Uri.file(getWorkspacePath()),
 		filters: {
-			'Data': ['RData']
-		}
+			'Data': ['RData'],
+		},
+		title : 'Load workspace'
 	}).then((uri: Uri[] | undefined) => {
 		if (uri) {
-			runTextInTerm(
-				`load(\"${(uri[0].fsPath.split(path.sep).join(path.posix.sep))}\")`
+			void runTextInTerm(
+				`load("${(uri[0].fsPath.split(path.sep).join(path.posix.sep))}")`
 			);
 		}
 	});
 }
+
+function getWorkspacePath(): string {
+		if (workspace.workspaceFolders) {
+			return workspace.workspaceFolders?.map(folder => folder.uri.path)[0]
```

Comment:
Since we support multi-root workspace, we should find the workspace folder of the active document.

You could use `workspace.getWorkspaceFolder(window.activeTextEditor.document.uri)` to get which workspace folder the active document belongs to.

The logic should be:

1. There's an active document
    1. Find its workspace folder
    2. If no workspace folder is found, then find its file folder via  `path.dirname(window.activeTextEditor.document.uri.fsPath)`.
    3. If it is not a file (e.g. untitled  document), use `os.homedir()`.
2. There's no active document.
    1. If there is a workspace folder, use the first one.
    2. If there is no open workspace folder, use `os.homedir()`.

Reply:
Wait, the workspace viewer gets data from `globalenv.json` from session watcher. I think it should use the session working directory.

Reply:
Do you mean the temporary folder housing the globalenv.json? E.g. "C:\Users\elian\AppData\Local\Temp\Rtmp2fxYLc\vscode-R"

Reply:
No, I mean the working directory of the session, which is known at `request.wd` (https://github.com/Ikuyadeu/vscode-R/blob/master/src/session.ts#L502) but we don't store it in a variable in `session.ts` yet.

Reply:
You could add an exported variable (e.g. `workingDir`) in `session.ts` and set the variable on session attach (e.g. after https://github.com/Ikuyadeu/vscode-R/blob/master/src/session.ts#L512), and use this variable instead of any workspace folder detection.